aws/aws.go

```diff
@@ -29,6 +33,96 @@ const subscriptionIDMaxLength = 40
 // convention we're limited to the total 80-character max of the SQS name length.
 const topicNameMaxLength = 40 - len(topicNamePrefix)
 
+// VerifyPermissions checks if the aws config exists and checks if it has permissions to
+// create sns topics, send messages, create SQS topics, delete topics, and delete sqs queues
+func VerifyPermissions(cfg *aws.Config) error {
+
+	// Check if environment contains aws config
+	_, errConfig := cfg.Credentials.Get()
+	if errConfig != nil {
+		return errConfig
+	}
+
+	topic := fmt.Sprintf("verifyPermissions")
+	subscriptionID := "verifySubsId"
+
+	s, subErr := NewSubscriber(cfg, topic, subscriptionID)
+	if subErr != nil {
+		return errors.New("error, failed to create a SQS queue: " + subErr.Error())
+	}
+
+	publisher, pubErr := NewPublisher(cfg, topic)
+	if pubErr != nil {
+		return errors.New("error, failed to create a SNS topic: " + pubErr.Error())
+	}
+
+	return verifyPermissions(s, publisher, topic, subscriptionID)
+}
+
+// verifyPermissions checks if the aws config has correct permissions
+func verifyPermissions(subscriber pubsub.Subscriber, publisher pubsub.Publisher, topic string, subID string) error {
+	defer func() {
+		// Delete the publisher and subscriber queue
+		DeletePublisherTopic(topic)
+		DeleteSubscriberByTopic(topic, subID)
+	}()
+
+	ctx, stop := context.WithTimeout(context.Background(), 1*time.Second)
+	defer stop()
+
+	c, e := subscriber.Start(ctx, nil)
+
+	testMessage := []byte("Permissions Verification Test Message.")
+	errSend := publisher.Publish(ctx, testMessage, nil)
+	if errSend != nil {
+		return errors.New("failed to push message " + errSend.Error())
+	}
+
+	select {
+	case msg, isOpen := <-c:
+		if !isOpen {
+			return errors.New("error, failed to open message channel")
+		}
+		if bytes.Equal(msg.Message(), testMessage) {
+			return nil
+		}
+		return errors.New("error, failed to receive message from publisher")
+	case err := <-e:
+		return err
+	}
+
+}
+
+// DeletePublisherTopic deletes the SNS topic
+func DeletePublisherTopic(topic string) error {
+	svc := sns.New(session.New())
+
+	paramsDelete := &sns.DeleteTopicInput{
+		TopicArn: aws.String("arn:aws:sns:us-east-1:405093580753:pubsub__" + topic),
```

Comment:
it's not recommended to construct the TopicArn like this: you should instead get the topic arn with a service call and use that.